executor/common_linux.h

```diff
@@ -5830,3 +5830,64 @@ static long syz_pidfd_open(volatile long pid, volatile long flags)
 }
 
 #endif
+
+#if SYZ_EXECUTOR || __NR_syz_ksmbd_send_req
+#include <netinet/in.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#define KSMBD_BUF_SIZE 16000
+
+static long syz_ksmbd_send_req(volatile long a0, volatile long a1, volatile long a2, volatile long a3)
```

Comment:
Do you plan to read out the ksmdb responses and somehow act upon them?

If not, we probably don't really need to introduce a pseudo system call at all. If you pass the kcov remote reference the standard way, all that remains are:
* A `socket` call.
* A `connect` call with the specific arguments.
* A `write` call.

These can be expressed in syzkaller descriptions and e.g. tied together by defining some custom `resource sock_ksmbd[sock]` resource.

Reply:
+1 for not adding a pseudo syscall. We just need to describe the tcp address, and add a special resource for the socket to tie all syscall together (socket/connect/sendmsg).

It has an additional advantage of giving more freedom to the fuzzer and exploring more scenarios. E.g. sending messages in parallel, calling shutdown, etc.